scalding-core/src/main/scala/com/twitter/scalding/Config.scala

```diff
@@ -51,6 +51,20 @@ trait Config extends Serializable {
     }
 
   /**
+   * Add files to be localized to the config. Intended to be used by user code.
+   * @param cachedFiles CachedFiles to be added
+   * @return new Config with cached files
+   */
+  def addDistributedCacheFiles(cachedFiles: CachedFile*): Config =
```

Comment:
No but the DistributedCacheFile class just doesn't make sense in the context of Executions. It has to be called from code that is intended for use only in a Job. Even getting access to the Execution's mode does't help as the mutations are ignored. I really think we need to not conflate these together.